status/discover.go

```diff
@@ -13,6 +16,73 @@ var saToken string
 
 var clientFactory kubernetes.ClientFactory
 
+func getClient() (kubernetes.IstioClientInterface, error) {
+	if saToken == "" {
+		token, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+		if err != nil {
+			return nil, err
+		}
+		saToken = string(token)
+	}
+
+	if clientFactory == nil {
+		userClientFactory, err := kubernetes.GetClientFactory()
+		if err != nil {
+			return nil, err
+		}
+		clientFactory = userClientFactory
+	}
+
+	client, err := clientFactory.GetClient(saToken)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the client is not openshift return and avoid discover
+	if !client.IsOpenShift() {
+		return nil, errors.New("Client is not Openshift")
+	}
+
+	return client, nil
+}
+
+func checkIfQueryBasePath(ns string, service string) (path string, err error) {
+	path = ""
+	client, err := getClient()
+
+	if err != nil {
+		return path, err
+	}
+	svc, err := client.GetService(ns, service)
+	if err != nil {
+		return path, err
+	}
+	labelSelectorService := labels.Set(svc.Labels).String()
```

Comment:
I think the correct way to fetch deployments from a service is fetch it from service selector (svc.Spec.Selector) instead of service labels. 

https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service

Reply:
IF we check the spec we can't find the deployment :(